nerdlets/tag-improver-nerdlet/components/tag-entity-view.js

```diff
@@ -74,27 +129,113 @@ export default class TagEntityView extends React.Component {
+      case '<show all entities>':
+        entityDisplayOption = DISPLAY_OPTION.ALL_ENTITIES;
+        break;
+      default:
+        entityDisplayOption = DISPLAY_OPTION.SPECIFIC_TAG_VALUE;
+    }
+
+    this.setState({
+      dropDownSelectedTagValue,
+      entityDisplayOption
+    });
+  };
+
+  addEntity = (entity, activeTagValue) => {
+    return {
+      entityName: entity.name,
+      entityGuid: entity.guid,
+      firstTagValue: activeTagValue,
+      complianceScore: entity.complianceScore,
+      mandatoryTags: entity.mandatoryTags,
+      optionalTags: entity.optionalTags
+    };
+  };
+
   getTableData = () => {
-    const { tagHierarchy } = this.props;
-    const { firstTagKey } = this.state;
+    const {
+      tagHierarchy,
+      entityTagsMap,
+      selectedTagKey,
+      selectedTagValue
+    } = this.props;
+    const {
+      firstTagKey,
+      entityDisplayOption,
+      dropDownSelectedTagValue
+    } = this.state;
+
+    let newTagHierarchy = {};
+    let tagKey = '';
+    let tagValue = '';
+    if (entityDisplayOption === DISPLAY_OPTION.SPECIFIC_TAG_VALUE) {
+      if (
+        selectedTagKey &&
+        selectedTagValue &&
+        selectedTagValue !== '<tag not defined>'
+      ) {
+        tagKey = selectedTagKey;
+        tagValue = selectedTagValue;
+      }
+      if (dropDownSelectedTagValue) {
+        tagKey = firstTagKey;
+        tagValue = dropDownSelectedTagValue;
+      }
+    }
+
+    if (tagKey && tagValue) {
+      newTagHierarchy[tagKey] = {};
+      newTagHierarchy[tagKey][tagValue] = tagHierarchy[tagKey][tagValue];
+    } else {
+      newTagHierarchy = tagHierarchy;
+    }
 
     const entities = {};
-    for (const tagData of Object.values(tagHierarchy)) {
+    for (const tagData of Object.values(newTagHierarchy)) {
       for (const entityList of Object.values(tagData)) {
         for (const entity of entityList) {
           if (!entities[entity.guid]) {
-            entities[entity.guid] = {
-              entityName: entity.name,
-              entityGuid: entity.guid,
-              firstTagValue: this.findTagValue(entity, firstTagKey),
-              complianceScore: entity.complianceScore,
-              mandatoryTags: entity.mandatoryTags,
-              optionalTags: entity.optionalTags
-            };
+            const activeTagValue =
+              tagValue || this.findTagValue(entity, firstTagKey);
+
+            const tagHasValue = entityTagsMap[entity.guid].find(
+              tag => tag.tagKey === firstTagKey
+            );
+
+            if (
+              ((entityDisplayOption === DISPLAY_OPTION.SPECIFIC_TAG_VALUE ||
+                entityDisplayOption === DISPLAY_OPTION.ALL_TAG_VALUES) &&
+                !tagHasValue) ||
+              (entityDisplayOption === DISPLAY_OPTION.TAG_NOT_DEFINED &&
+                tagHasValue)
+            ) {
+              continue;
+            } else {
+              // if (selectAllEntities && selectedTagKey && selectedTagValue) {
```

Comment:
@shahramk is this code needed or should it be deleted?

Reply:
removed dead code.